lib/Dialect/FIRRTL/Transforms/LowerTypes.cpp

```diff
@@ -774,8 +824,83 @@ void FIRRTLTypesLowering::getAllBundleLowerings(
       .Default([&](auto) {});
 }
 
-Identifier FIRRTLTypesLowering::getArgAttrName(unsigned newArgNumber) {
+Identifier TypeLoweringVisitor::getArgAttrName(unsigned newArgNumber) {
   SmallString<16> argNameBuffer;
   mlir::impl::getArgAttrName(newArgNumber, argNameBuffer);
-  return Identifier::get(argNameBuffer, &getContext());
+  return Identifier::get(argNameBuffer, context);
+}
+
+//===----------------------------------------------------------------------===//
+// Pass Infrastructure
+//===----------------------------------------------------------------------===//
+
+namespace {
+struct LowerTypesPass : public LowerFIRRTLTypesBase<LowerTypesPass> {
+  void runOnOperation() override;
+
+private:
+  void runAsync();
+  void runSync();
+  // A list of visitors, one for each thread.  These are cached by the pass.
+  std::vector<TypeLoweringVisitor> visitors;
```

Comment:
Does visitors need to be maintained as state in the pass?  Can't each instance be something in the `perThread` lambda?  Constructing these things are effectively free.  This also eliminates the resize and dependence on compute_thread_count

Reply:
I made a mistake reading the git diff. The existing code likes to reuse the vectors, caching them in the Pass (now Visitor) and reseting them before running again.  It seemed like it might be a micro-optimization to reduce the number of times we potentially call `malloc`, but maybe its just incidental.  I'll stop doing that, since it makes everything complicated.

Reply:
Fixed! No longer caching this.